source/common/network/connection_impl.cc

```diff
@@ -387,6 +389,9 @@ void ConnectionImpl::write(Buffer::Instance& data, bool end_stream) {
     // with a connection error if a call to write(2) occurs before the connection is completed.
     if (!connecting_) {
       ASSERT(file_event_ != nullptr, "ConnectionImpl file event was unexpectedly reset");
+      if (file_event_ == nullptr) {
```

Comment:
I don't think we want this, given the ASSERT above this says this should never happen. Can you describe more what you are trying to solve?